app.cr

```diff
@@ -14,10 +14,26 @@ def headers
   headers
 end
 
-def crystal_repos(sort)
+def crystal_repos
   client = HTTP::Client.new("api.github.com", 443, true)
-  response = client.get("/search/repositories?q=language:crystal&per_page=100&sort=#{sort}", headers)
-  GithubRepos.from_json(response.body)
+  response = client.get("/search/repositories?q=language:crystal&per_page=100", headers)
+  total_repos = GithubRepos.from_json(response.body).total_count
+  repos = GithubRepos.from_json(response.body)
+
+  max_pages = (total_repos.to_f / 100).round.to_i
+  max_pages.times do |i|
```

Comment:
You are totally right, triggering this by a real user is not very smart. :(
The problem is see with this is that if we don't have all repositories in our cache (if no one went through all the pages in the last 30 minutes), we can't use filtering for our search. We also can't use the github api for search because its limited to 10 requests per minute. 

Something we could do until we find a good data persistence solution is to have a route that refreshes the cache (e.g ``get "/refresh_cache"``) that executes all these requests and trigger it via the heroku scheduler every 30 minutes to keep our cache up to date.